core-v1/src/main/scala/dsl/BuildExprDsl.scala

```diff
@@ -32,6 +32,22 @@ trait BuildExprDsl extends DebugExprDsl with WrapArityMethods {
 
   def ident[I](implicit opI: OP[W[I]]): Expr.Identity[W[I], OP]
 
+  final def concat[C[_] : Foldable, I, A](
+    expressions: I ~:> C[A]*,
+  )(implicit
+    opSSA: OP[Seq[Seq[A]]],
+    opSA: OP[Seq[A]],
+  ): AndThen[I, Seq[Seq[A]], Seq[A]] = {
+    Expr
+      .Sequence(expressions.map {
+        Expr.Select(_, VariantLens.id[C[A]].asIterable.to[Seq], (_: C[A], sa: Seq[A]) => sa)
+      })
+      .andThen(Expr.Flatten())
+  }
+
+  def flatten[I, C[+_] : FlatMap, A](expr: I ~:> C[C[A]])(implicit opCA: OP[C[A]]): AndThen[I, C[C[A]], C[A]] =
+    expr.andThen(Expr.Flatten())
```

Comment:
I think this should probably be a method on the higher kinded type, rather than a top-level method.